### PR TITLE
Features, tests and stuff

### PR DESCRIPTION
**Pull request by tuomas777**
* Api key authentication
* Permission checking to operator API
* Limited the time parkings are editable after creation
* Parking create/update/delete tests
* Parking status and registration code filters to internal API
* Address as a nested object in both APIs
* A bunch of minor enhancements

### CODE REVIEW

File: parkings/api/operator/parking.py [hunk 1]
```diff
@@ -1,14 +1,101 @@
...
+            pass
+
+        return False
+
+    def has_object_permission(self, request, view, obj):
+        """
+        Allow only operators to modify and only their own parkings and
+        only for a fixed period of time after creation.
+        """
+        user = request.user
```
Comment (jaywink) [inline parkings/api/operator/parking.py]: Most of the code here is using the same as `has_permission` above. Maybe get the result of that first and return if `False` before doing the last operator equals and time checks relevant to this method?

File: parkings/models/parking.py [hunk 2]
```diff
@@ -29,17 +32,29 @@ class Parking(TimestampedModelMixin, UUIDPrimaryKeyMixin):
...
     def __str__(self):
         start = localtime(self.time_start).replace(tzinfo=None)
         end = localtime(self.time_end).time().replace(tzinfo=None)
         return "%s -> %s (%s)" % (start, end, self.registration_number)
+
+    def is_valid(self):
+        return self.time_start <= now() <= self.time_end
+
+    def get_state(self):
+        if self.time_start <= now() <= self.time_end:
```
Comment (jaywink) [inline parkings/models/parking.py]: Less code duplication with `if self.is_valid():`.

Also, what about making `is_valid` a property and `get_state` also a property, probably called `state`?

File: parkkihubi/settings.py [hunk 3]
```diff
@@ -154,6 +156,14 @@
...
+    'DEFAULT_FILTER_BACKENDS': ('django_filters.rest_framework.DjangoFilterBackend',),
     'PAGE_SIZE': 20,
     'TEST_REQUEST_DEFAULT_FORMAT': 'json',
 }
+
+
+##############
+# Parkkihubi #
+##############
+TIME_PARKINGS_EDITABLE = timedelta(minutes=2)
```
Comment (jaywink) [inline parkkihubi/settings.py]: Personally I prefer to prefix settings with their app context, so for example `PARKINGS_TIME_EDITABLE`. This makes it easier to see what context the setting relates to. Leave it up to you.

File: requirements.in [hunk 4]
```diff
@@ -13,6 +13,8 @@ psycopg2
 # Django REST Framework
 djangorestframework
 djangorestframework-gis
+django-filter
 
 # Misc
 pytz
+freezegun
```
Comment (jaywink) [inline requirements.in]: This should probably be in dev requirements only?

File: parkings/api/operator/parking.py (hunk 1 above)
Comment (tuomas777) [inline parkings/api/operator/parking.py]: Actually I think that `has_permission()` will be called automatically before `has_object_permission()` anyway :) This needs some tuning.

File: parkings/models/parking.py (hunk 2 above)
Comment (tuomas777) [inline parkings/models/parking.py]: Actually `is_valid()` is a leftover from a previous implementation and should not be there at all :D 

About properties, personally I prefer methods in these where the value needs to be calculated from other values to distinguish from normal attributes.

File: parkkihubi/settings.py (hunk 3 above)
Comment (tuomas777) [inline parkkihubi/settings.py]: Yep that is good, changing.

File: requirements.in (hunk 4 above)
Comment (tuomas777) [inline requirements.in]: Yep.

**Comment by tuomas777** [comment]
Fixed the reported issues.

**Pull request merged**